### PR TITLE
SSL Support

### PR DESCRIPTION
**Pull request by estebgonza**
Support SSL communication: Client <=SSL=> Maggregor <=SSL=> MongoDB.

New parameters:
```
# SSL
USE_SSL=false
SSL_KEY_PATH=
SSL_CERT_PATH=
``` 
For e2e remote scenarios  (Atlas URL with credentials):
```
export TEST_ATLAS_URL=
```

**Bonus:**
- Fix in the CacheService: unsubscribe was not being called.
- Remove the retry limit in ListenerService: we want to keep attempting to reconnect indefinitely.

### CODE REVIEW

File: src/server/modules/mongodb-listener/listener.service.ts [hunk 1]
```diff
@@ -45,20 +46,22 @@ export class ListenerService extends EventEmitter {
...
-        } else {
-          this.logger.warn(
-            'Failed to connect: retrying to connect to MongoDB in 1 second...',
-          );
-          await new Promise((resolve) => setTimeout(resolve, 1000));
-        }
+        this.logger.warn(
+          `Failed to connect: retrying to connect to MongoDB in 1 second... Error: ${error.message}`,
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (NicoGGG) [inline src/server/modules/mongodb-listener/listener.service.ts:64]: This should be tested in integration test listener.test.ts. Why is it not the case ?

File: src/server/modules/mongodb-proxy/proxy.service.ts [hunk 2]
```diff
@@ -81,8 +82,20 @@ export class MongoDBTcpProxyService extends EventEmitter {
...
+      };
+
+      const proxySocket = this.sslOptions
+        ? tls.connect({
+            ...connectOptions,
+            servername: this.options.targetHost,
+            rejectUnauthorized: false,
+          })
+        : net.connect(connectOptions);
+
```
Comment (NicoGGG) [inline src/server/modules/mongodb-proxy/proxy.service.ts:98]: Do we assume that mongodb has to be SSL if maggregor is ? Does it even work ?

File: tests/e2e/contexts.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+} & CommonContext;
+
+export type Context = RemoteContext | LocalContext;
+
+export const contexts: Context[] = [
+  {
+    type: 'remote',
+    name: 'Remote on shared MongoDB Atlas v6.0.0',
+    ssl: true,
+    externalMongoUri: `mongodb://${testAtlasUsername}:${testAtlasPassword}@ac-edqjyiy-shard-00-02.puyjt1x.mongodb.net:27017/?ssl=true&replicaSet=atlas-9duftq-shard-0&retryWrites=true&w=majority`,
```
Comment (NicoGGG) [inline tests/e2e/contexts.ts:45]: the hostname should also be an env variable in my opinion

**Review by NicoGGG — APPROVED** [review]

File: tests/e2e/contexts.ts (hunk 3 above)
Comment (NicoGGG) [reply tests/e2e/contexts.ts:45]: There is also the replicatSet. So yes maybe just use a variable for the entire url

File: src/server/modules/mongodb-listener/listener.service.ts (hunk 1 above)
Comment (estebgonza) [reply src/server/modules/mongodb-listener/listener.service.ts:64]: This part was not backed by an integration test. I did it.

**Pull request merged**